src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -108,8 +115,22 @@ private sealed partial class TagSource : ForegroundThreadAffinitizedObject
 
                 Connect();
 
-                // Kick off a task to compute the initial set of tags.
-                RecalculateTagsOnChanged(new TaggerEventArgs(TaggerDelay.Short));
+                // Kick off a task to immediately compute the initial set of tags. This work should 
+                // not be cancellable (except if we get completely released), even if more events come 
+                // in.  That way we can get the initial set of results for the buffer as quickly as 
+                // possible, without kicking the work  down the road.
+                var initialTagsCancellationToken = _initialComputationCancellationTokenSource.Token;
+                if (_workQueue.IsForeground())
+                {
+                    RecomputeTagsForeground(cancellationTokenOpt: initialTagsCancellationToken);
+                }
+                else
+                {
+                    RegisterNotification(
+                        () => RecomputeTagsForeground(cancellationTokenOpt: initialTagsCancellationToken),
```

Comment:
Not without a large rewrite of tagging.  The UI thread acts as a sync point for us to ensure proper state introspection and mutation.  We try to keep that work to a minimum, doing the actual computation on the BG thread.  The UI thread is used to just store state in a consistent fashion so we can not lock when things like GetTags are called.